chore: monorepo setup

snowbytes opened the pull request:
# chore: monorepo setup

## Description

This PR closes [WRKSY-33](https://v58-tier3-team-33.atlassian.net/browse/WRKSY-33). 

## Review Checklist

- [x] [WRKSY-33](https://v58-tier3-team-33.atlassian.net/browse/WRKSY-33)'s acceptance criteria are validated.
- [x] solution adheres to our [Definition of Done](https://github.com/chingu-voyages/V58-tier3-team-33/wiki/Definition-of-Done).
- [x] No regressions, new warnings or errors introduced.

snowbytes committed:
chore(setup): initialize barebones monorepo

Co-authored-by: asiill <9745019+asiill@users.noreply.github.com>

snowbytes committed:
chore(client): init the web frontend using vite

Co-authored-by: asiill <9745019+asiill@users.noreply.github.com>

asiill committed:
chore(server): init the node server

Co-authored-by: snowbytes <84378360+snowbytes@users.noreply.github.com>

snowbytes committed:
chore(server): add typescript support

rachel-labri-tipton reviewed (dismissed):
This looks good to me. I was able to run both the web app and the server using the information in the updated docs and the config files look good as well. 

Great work @snowbytes and @asiill!

AskTiba reviewed (commented):
I have a concern with the folder name of web in the apps directory, the first impression i got was that it is a purely frontend web based app. How about we name it client instead.

Everything else checked out okay with me. Great work.

snowbytes commented:
> I have a concern with the folder name of web in the apps directory, the first impression i got was that it is a purely frontend web based app. How about we name it client instead.

@AskTiba I don't understand what the concern is, sorry about that. Could you give me more details? What I understand is that app (`apps/web`) is a pure frontend web app. while `apps/server` is our backend server.

AskTiba commented:
> > I have a concern with the folder name of web in the apps directory, the first impression i got was that it is a purely frontend web based app. How about we name it client instead.
> 
> @AskTiba I don't understand what the concern is, sorry about that. Could you give me more details? What I understand is that app (`apps/web`) is a pure frontend web app. while `apps/server` is our backend server.

Using web instead of client is a concern because it restricts the naming to one platform and creates confusion later when additional frontends—like mobile or desktop—are introduced.

snowbytes commented:
> Using web instead of client is a concern because it restricts the naming to one platform and creates confusion later when additional frontends—like mobile or desktop—are introduced.

Signaling that it's web based was intentional on my end as I believe it *clears* any confusion about the nature of the frontend, but I acknowledge that for our project, the distinction between web vs client is meaningless as we won't have any additional specialized frontends down the line (like native desktop, mobile etc..). I will change it per your recommendation. Thanks for the feedback!

snowbytes committed:
feat(server): add db config & a health module

snowbytes committed:
chore(client): add tailwind

snowbytes commented:
@rachel-labri-tipton I switched back to draft momentarily to address @AskTiba's feedback, but then realized that I missed adding both tailwind and postgres/kysely. I edited the Jira ticket to require all of our production tech stack in the acceptance criteria. Checklist of what I skipped before:
- postgres + kysely setup
- tailwind setup

I also went ahead and added a minimal `health` endpoint that returns status of services (so far only db and overall status) as a way to test that the db properly integrates. Readme was updated to include the dev setup updates, but now also including a list of external dependencies needed (pgadmin and postgres for local setup so far).

Thank you for the reviews in advance

rachel-labri-tipton reviewed (approved):
Hi @snowbytes, 
Great work here with clear supporting documentation. I had assumed yesterday that we'd layer tailwind and backend TS support in after the core foundation for the app was set so didn't ask about it. 

I'm not familiar with Kysely. Looks like they've got clear documentation. 

I just set things up locally using your new changes to the docs. Looks like things are working. 
Please note, I didn't spin up a database instance but the health check failed, which is what it should do if no database is running.

snowbytes commented:
Hi @rachel-labri-tipton! The setup for kysely is not what they have in the docs as I made it modular in a way that respects the principle of least privilege whereas they expose all tables in a centralized fashion. I'd be happy to go over it with you if you ever need a quick explanation anytime.

Glad everything was clear for you. Thanks for being reactive 😄
